### PR TITLE
Setup Storybook

### LINKED ISSUE #31: Setup Storybook
https://storybook.js.org/docs/react/get-started/install

**NB:** must works with themes.

### PR DESCRIPTION
**Pull request by peter-nikitin**
Add storyboard with theme support 

Resolve #31

### CODE REVIEW

**Comment by peter-nikitin** [comment]
There is `RangeError: invalid time interval` in  `EstimateDropdown`. Can't handle it properly

File: src/stories/Button/Button.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+
+import { Button } from './Button';
+
+// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
+export default {
+    title: 'TaskAny/Button',
```
Comment (awinogradov) [inline src/stories/Button/Button.stories.tsx:8]: Only first letter capitalized

File: src/stories/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from '../../components/Button';
```
Comment (awinogradov) [inline src/stories/Button/Button.tsx:1]: You can import `Button` directly w/o re-export

File: .storybook/preview.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { GeistProvider, CssBaseline, Themes } from '@geist-ui/core';
+import { ThemeProvider } from 'next-themes';
+import { backgroundColor } from '../src/design/@generated/themes'; //design/@generated/themes
```
Comment (awinogradov) [inline .storybook/preview.js:4]: Useless comment here. Same info in the import string.

File: .storybook/preview.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        matchers: {
+            color: /(background|color)$/i,
+            date: /Date$/,
+        },
+    },
+};
+
+export const decorators = [
+    (Story, context) => {
+
```
Comment (awinogradov) [inline .storybook/preview.js:22]: Useless and empty 2 lines

**Review by awinogradov — CHANGES_REQUESTED** [review]
OMG! @peter-nikitin thanks for this contribution! 🔥 

I found some small mistakes. It must be fixed before we can merge to main.

Also, I have question about assets. How are you using them? Do we really need this icons in the project?

**Comment by awinogradov** [comment]
Some recommendation about PR. Usually we use one PR for one issue. In this case would be better to use same title with issue for PR. Thanks 🙏🏼

File: src/stories/Button/Button.stories.tsx (hunk 1 above)
Comment (peter-nikitin) [reply src/stories/Button/Button.stories.tsx]: Fix

File: .storybook/preview.js (hunk 3 above)
Comment (peter-nikitin) [reply .storybook/preview.js]: Fix

File: .storybook/preview.js (hunk 4 above)
Comment (peter-nikitin) [reply .storybook/preview.js]: Fix

**Comment by awinogradov** [comment]
> There is RangeError: invalid time interval in EstimateDropdown. Can't handle it properly

I think you can remove/ignore this story right now. Some of completions and dropdowns have requests to API. It must be mocked in examples. 

> Can start storybook without src/stories/Button/Button.tsx. Fails on build

Can you show error stacktrace?

File: src/stories/Button/Button.tsx (hunk 2 above)
Comment (peter-nikitin) [reply src/stories/Button/Button.tsx]: Fix

**Comment by peter-nikitin** [comment]
>> Can start storybook without src/stories/Button/Button.tsx. Fails on build
> Can you show error stacktrace?

Disappear )

**Comment by peter-nikitin** [comment]
> Also, I have question about assets. How are you using them? Do we really need this icons in the project?

Remove. They were from boilerplate

**Comment by peter-nikitin** [comment]
https://github.com/taskany-inc/issues/issues/31

**Comment by awinogradov** [comment]
> [peter-nikitin](https://github.com/peter-nikitin) changed the title Issues/31 Setup Storybook #31

So close... But ;) Now PR's title contains issue id and PR id.

image

Little bit controversial...

So, you must remove #31 from the title and add it to PR's description, like `Resolve #31`.

File: .storybook/main.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        fileLoaderRule.exclude = /\.svg$/;
+
+        rules.push({
+            test: /\.svg$/,
+            use: ["@svgr/webpack"],
+        });
+
+        return config;
+    },
+    "stories": [
```
Comment (awinogradov) [inline .storybook/main.js:14]: Strange to see simple keys in quotes. But I think this is problem with formatter.

**Review by awinogradov — APPROVED** [review]

**Pull request merged**